getting_started/step_by_step/resources.rst

```diff
@@ -158,12 +158,173 @@ Freeing resources
 Resource extends from :ref:`Reference <class_Reference>`.
 As such, when a resource is no longer in use, it will automatically free
 itself. Since, in most cases, Resources are contained in Nodes, scripts
-or other resources, when a node is removed or freed, all the children
+or other resources, when a node is removed or freed, all the owned
 resources are freed too.
 
 Scripting
 ---------
 
-Like any object in Godot, not just nodes, resources can be scripted,
-too. However, there isn't generally much of an advantage, as resources
-are just data containers.
+Like any Object in Godot, users can also script Resources. Resource
+scripts grant all the same benefits as the base Resource object. They
+inherit both reference-counting memory management and automatic serialization
+between binary or text data (\*.res, \*.tres) and in-memory object
+properties.
+
+This comes with many distinct advantages over alternative data
+structures such as JSON, CSV, or custom TXT files. Users can only import these 
+assets as a :ref:`Dictionary <class_Dictionary>` (JSON) or as a 
+:ref:`File <class_File>` to parse. What sets Resources apart is their 
+inheritance of :ref:`Object <class_Object>`, :ref:`Reference <class_Reference>`,
+and :ref:`Resource <class_Resource>` features:
+
+- They can define constants, so constants from other data fields or objects are not needed.
+
+- They can define methods, including setter/getter methods for properties. This allows for abstraction and encapsulation of the underlying data. If the Resource script's structure needs to change, the game using the Resource need not also change.
+
+- They can define signals, so Resources can trigger responses to changes in the data they manage.
+
+- They have defined properties, so users know 100% that their data will exist.
+
+- Resource auto-serialization and deserialization is a built-in Godot Engine feature. Users do not need to implement custom logic to import/export a resource file's data.
+
+- Resources can even serialize sub-Resources recursively, meaning users can design even more sophisticated data structures.
+
+- Users can save Resources as version-control-friendly text files (\*.tres). Upon exporting a game, Godot serializes resource files as binary files (\*.res) for increased compression.
```

Comment:
Aren't `*.res` files also faster to load?

Reply:
Yes, because the same amount of data is packed into a smaller space, so it doesn't take as long to parse the data as input.

Reply:
I suppose I could do "for increased speed and compression" instead.